Userland/Services/SystemServer/main.cpp

```diff
@@ -498,6 +520,42 @@ static ErrorOr<void> reopen_base_file_descriptors()
     return {};
 }
 
+static ErrorOr<void> activate_base_services_based_on_system_mode()
+{
+    if (g_system_mode == "graphical") {
+        bool found_gpu_device = false;
+        for (int attempt = 0; attempt < 10; attempt++) {
+            struct stat file_state;
+            int rc = lstat("/dev/gpu/connector0", &file_state);
+            if (rc == 0) {
+                found_gpu_device = true;
+                break;
+            }
+            sleep(1);
+        }
+        if (!found_gpu_device) {
+            dbgln("WARNING: No device nodes at /dev/gpu/ directory after 10 seconds. This is probably a sign of disabled graphics functionality.");
+            dbgln("To cope with this, graphical mode will not be enabled.");
+            g_system_mode = "text";
```

Comment:
Instead of coming up with constants like these on the fly, we should probably have an enum or constexprs somewhere to reference.